buildchain/buildchain/iso.py

```diff
@@ -188,21 +189,36 @@ def action(targets: Sequence[str]) -> None:
 @doit.create_after(executed='populate_iso')  # type: ignore
 def task__iso_build() -> types.TaskDict:
     """Create the ISO from the files in ISO_ROOT."""
-    mkisofs = [
-        config.ExtCommand.MKISOFS.value, '-output',  ISO_FILE,
-        '-quiet',
-        '-rock',
-        '-joliet',
-        '-joliet-long',
-        '-full-iso9660-filenames',
-        '-volid', '{} {}'.format(config.PROJECT_NAME, versions.VERSION),
-        '--iso-level', '3',
-        '-gid', '0',
-        '-uid', '0',
-        '-input-charset', 'utf-8',
-        '-output-charset', 'utf-8',
-        constants.ISO_ROOT
-    ]
+    def unlink_if_exist(filepath: Path) -> None:
+        """Delete a file if it exists."""
+        try:
+            filepath.unlink()
+        except FileNotFoundError:
+            pass
+
+    def mkisofs() -> None:
+        """Create an ISO file (delete on error)."""
+        cmd = [
+            config.ExtCommand.MKISOFS.value, '-output',  ISO_FILE,
+            '-quiet',
+            '-rock',
+            '-joliet',
+            '-joliet-long',
+            '-full-iso9660-filenames',
+            '-volid', '{} {}'.format(config.PROJECT_NAME, versions.VERSION),
+            '--iso-level', '3',
+            '-gid', '0',
+            '-uid', '0',
+            '-input-charset', 'utf-8',
+            '-output-charset', 'utf-8',
+            constants.ISO_ROOT
+        ]
+        try:
+            subprocess.run(cmd, check=True)
+        except:
```

Comment:
Can't we just have `except CalledProcessError`? We only want to catch errors that occurred during our `run`, since they're the only ones that _may_ generate partial artifacts we want to clean up.

Reply:
Since the cleanup handle the case of "missing file" I guess we can always cleanup, to be on the safe side.